lib/puppet/util/yaml.rb

```diff
@@ -9,6 +9,17 @@ module Puppet::Util::Yaml
 
   class YamlLoadError < Puppet::Error; end
 
+  if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.1.0')
+    def self.safe_load(yaml, allowed_classes = [], filename = nil)
+      YAML.safe_load(yaml, allowed_classes, [], false, filename)
```

Comment:
`safe_load` behaves differently than `load` if given an empty string. The former returns `nil`; the latter `false`, which breaks the lookup code:

```
(byebug) content
""
(byebug) path
"/var/folders/8m/d6146zxd6bv_3h8y4ykvvxh40000gn/T/rspecrun20180803-89228-1jx97au/environments20180803-89228-57pw23/spec/modules/mod/data/common.yaml"
(byebug) File.read("/var/folders/8m/d6146zxd6bv_3h8y4ykvvxh40000gn/T/rspecrun20180803-89228-1jx97au/environments20180803-89228-57pw23/spec/modules/mod/data/common.yaml")
""
(byebug) ::YAML.safe_load(content, [], [], nil, path)
nil
(byebug) ::YAML.load(content, path)
false
```